### PR TITLE
Single-argument shorthand syntax for CPMAddPackage

### LINKED ISSUE #205: Single-arg CPMAddPackage
This is a feature proposal. If you approve of it, I can implement in my spare time by the end of the week.

* Check if `CPMAddPackage` is called with a single argument.
* If yes, split it along "@". Interpret the first part as a git repo, and the second as a version
* If the git repo matches `<user>/<name>` or `gh:<user>/<name>` interpret it as a GitHub package
* If the git repo matches `gl:<user>/<name>` interpret it as a GitLab package
* Otherwise interpret it as a git repository uri

Thus some valid uses would include:

```cmake
CPMAddPackage(catchorg/Catch2@2.5.0)
CPMAddPackage(gl:graphviz/graphviz@2.46.1)
CPMAddPackage(https://example.com/myrepo.git@1.0.0)
```

### PR DESCRIPTION
**Pull request by iboB**

### CODE REVIEW

**Comment by iboB** [comment]
Apparently GitHub doesn't support crossrefs from titles, so adding this for posterity: 

This closes #205

**Comment by iboB** [comment]
Oops. There's a bug. The uri shouldn't be tolower-ed. Don't merge. Will fix several hours

**Comment by iboB** [comment]
Fixed. Ready to merge

File: test/unit/parse_add_package_single_arg.cmake
Comment (TheLartians) [inline test/unit/parse_add_package_single_arg.cmake:62]: These are currently unsupported test-cases right? Let's add a comment to document them as such.

```suggestion
# The following test cases are to be used in the future, once single-argument archives are supported

# cpm_parse_add_package_single_arg("https://example.org/foo.tar.gz" args)

# assert_equal("URL;https://example.org/foo.tar.gz" "${args}")

# cpm_parse_add_package_single_arg("https://example.org/foo.tar.gz#baadf00d@1.2.0" args)

# assert_equal("URL;https://example.org/foo.tar.gz;URL_HASH;baadf00d;VERSION;1.2.0" "${args}")

# cpm_parse_add_package_single_arg("ftp://user:password@server/pathname.zip#fragment#0ddb411@0"
# args)

# assert_equal("URL;ftp://user:password@server/pathname.zip#fragment;URL_HASH;0ddb411;VERSION;0"
# "${args}")
```

**Review by TheLartians — COMMENTED** [review]
Awesome, thanks for implementing the new concise syntax! I'll do some tests myself before merging, but looks great! 🚀

File: README.md
Comment (iboB) [reply README.md]: Yep! Good catch on the typo

File: README.md
Comment (iboB) [reply README.md:61]: Check. I can see how this default can be viewed as GitHub favoritism :)

I still think that some default is better than no default. Yes it will be difficult to change in the future, but until that future we will the most popular choice as a default. After that future, things will be undistinguishable from the no-default scenario. 

Anyway, I'll remove the gh default. We can always add it at a later point if we so decide.

**Comment by TheLartians** [comment]
FYI I've created a pre-release [`v0.31.0-preview-1`](https://github.com/cpm-cmake/CPM.cmake/releases/tag/v0.31.0-preview-1) for easy testing of the new syntax.

**Comment by TheLartians** [comment]
Oops, I selected the wrong branch and it turns out I can't reference external PRs in releases 🤦

**Review by TheLartians — APPROVED** [review]
I've done some testing and the feature works great so far! Install behaviour of CMake does seem to work as expected: if a project depends on an "EXCLUDED" dependency, the dependency is installed alongside.

I'd still like to do some updates on the examples and Readme as I see this becoming the standard way to add "well-behaved" dependencies, but that's something I can take care of in another PR.

**Pull request merged**
